torch/csrc/autograd/python_function.cpp

```diff
@@ -889,12 +895,28 @@ std::pair<UnpackedInput, InputFlags> unpack_input(PyObject* args) {
       }
       Py_INCREF(Py_False);
       PyTuple_SET_ITEM(flags.needs_input_grad.get(), i, Py_False);
+
+      if (profiler_need_input) {
+        // The following conversion from PyObject to IValue is expensive
+        // Only do it if profiler is enabled and needs input shapes
+        auto match = torch::jit::tryToInferPrimitiveType(arg);
+        if (match.success()) {
+          unpacked.record_function_inputs.push_back(
+              torch::jit::toIValue(arg, match.type()));
+        } else {
+          PyErr_WarnEx(
+              PyExc_UserWarning,
+              "Only primitive type (bool, float, int, complex) or its container (list, tuple, dict) is supported for non-tensor input in python function profiler",
+              0);
```

Comment:
I'm not sure we want this warning? It seems that in most usages that go through this path, we won't actually need the record_function_inputs to be recorded. The only case is where we want to do some kind of replay.

I think you should either have an explicit mode which you can enable that says I need to replay later, and if that mode is enabled, you should error immediately here.

The other option is to not do anything here, and just error when you actually go and do the replay.

Reply:
I removed it. It is not straight forward to know if user want to capture execution trace. @briancoutinho is working on combining Kineto trace and execution trace. After that change is in, it should be trivial to check it. I will update  profiler_need_input to profiler_need_replay after Brian's change.